### PR TITLE
Document modifications

### PR DESCRIPTION
**Pull request by Allenator1**
Please tell me if any of the theory is inaccurate - I mostly went off of Gemini and my discussions with you.

I changed the 'supported-theories' command back to 'supported-onnx-types' since the theories that the verifier supports are already covered under 'supported-logics'.

### CODE REVIEW

**Comment by Allenator1** [comment]
All done! One point is that netron won't show both the node name and the dimensions. I stuck to showing node names for the hidden node example, and set it to show dimensions for the other examples.

One question I had is regarding the terminology for 'hidden nodes'. Doing a bit of research into ONNX, it looks like each NodeProto stores a list of input and output names. Each node also has its own name attribute. In VNN-LIB, we would be interested in extracting one of the inputs/outputs to a node rather than the node itself (since a node can have more than one input/output). Therefore, rather than using the 'hidden node' terminology, would it be better to call it something else?

**Comment by MatthewDaggitt** [comment]
> All done! One point is that netron won't show both the node name and the dimensions. I stuck to showing node names for the hidden node example, and set it to show dimensions for the other examples.

Thanks looks great, can I ask for a few more minor tweaks?

- Figure 3.1 
   - change the ONNX node names to `input`, `output`
- Figure 3.2
   - change the names to `image`, `metadata`, `bbox` and `logits`, 
   - change the `MatMul` before `logits_out` to a `Softmax` (which actually produces logits!)
   - change the `bbox_out` dimensions to `2 x 4` (actually give pixel coordinates)
- Figure 3
   - change the ONNX node names to `input`, `output`
  -  change the `encoder_output` name to `hidden`
- Figure 4
  -  change the 

> Doing a bit of research into ONNX, it looks like each NodeProto stores a list of input and output names. Each node also has its own name attribute. In VNN-LIB, we would be interested in extracting one of the inputs/outputs to a node rather than the node itself

Thanks, this is an important point. Yes, we are referencing the output of the hidden node. I have added a TODO to the document.

> Therefore, rather than using the 'hidden node' terminology, would it be better to call it something else?

Hmm I could get on board with `declare-hidden-output`?

**Comment by MatthewDaggitt** [comment]
Ah also in Figure 4.1 please could you:
- add `(Real)` to the `R` node and `Arbitrary precision floating point`/`32 bit floating point` to the `FP` nodes?
- Add 64 bit, 8 bit and 4 bit floating point types.
- get rid of the `Orthogonal logic` category, and instead fold `SI` underneath the `Real` theory (maybe renaming it `SR (Reals with strict comparisons)`

If adding all the names in brackets all get a bit much, then you could remove them from the diagram and instead add a table underneath with the acronym and then a description?

**Comment by MatthewDaggitt** [comment]
Will move on to Chapter 2 tomorrow.

**Comment by Allenator1** [comment]
Hi Matthew, I thought bounding boxes only had 4 coordinates? E.g., [x, y, width, height]
